src/python_testing/TC_ICDM_5_1.py

```diff
@@ -0,0 +1,206 @@
+# test-runner-run/run1/app: ${LIT_ICD_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+
+import logging
+from dataclasses import dataclass
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mdns_discovery import mdns_discovery
+from mobly import asserts
+
+cluster = Clusters.Objects.IcdManagement
+commands = cluster.Commands
+attributes = cluster.Attributes
+modes = cluster.Enums.OperatingModeEnum
+ClientTypeEnum = cluster.Enums.ClientTypeEnum
+
+
+@dataclass
+class Client:
+    checkInNodeID: int
+    subjectId: int
+    key: bytes
+    clientType: ClientTypeEnum
+
+
+logger = logging.getLogger(__name__)
+kRootEndpointId = 0
+
+client1 = Client(
+    checkInNodeID=1,
+    subjectId=1,
+    key=b"\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f",
+    clientType=ClientTypeEnum.kEphemeral
+)
+
+
+class TC_ICDM_5_1(MatterBaseTest):
+
+    #
+    # Class Helper functions
+    #
+
+    async def _read_icdm_attribute_expect_success(self, attribute):
+        return await self.read_single_attribute_check_success(endpoint=kRootEndpointId, cluster=cluster, attribute=attribute)
+
+    async def _send_single_icdm_command(self, command):
+        return await self.send_single_cmd(command, endpoint=kRootEndpointId)
+
+    async def _get_icd_txt_record(self):
+        discovery = mdns_discovery.MdnsDiscovery()
+        localCompressedNodeId = hex(self.default_controller.GetCompressedFabricId())
+        dutNodeId = hex(self.dut_node_id)
+
+        services = await discovery.get_all_services(log_output=True, discovery_timeout_sec=240)
+        icdTxtRecord = modes.kUnknownEnumValue.value
+        print("debugging mdns")
+        print(services)
+        asserts.assert_true(mdns_discovery.MdnsServiceType.OPERATIONAL.value in services,
+                            "Could not find ICD operational service information.")
+
+        for service in services[mdns_discovery.MdnsServiceType.OPERATIONAL.value]:
+
+            # Generate hex for the compressedFabric and NodeId
+            serviceName = service.instance_name.split("-")
+            compressedFabric = hex(int(serviceName[0], 16))
+            nodeId = hex(int(serviceName[1], 16))
+
+            # Find the correct MdnsServiceEntry
+            if localCompressedNodeId == compressedFabric and dutNodeId == nodeId:
+                icdTxtRecord = service.txt_record['ICD']
```

Comment:
```suggestion
                icdTxtRecord = OperatingModeEnum(int(service.txt_record['ICD']))
                if icdTxtRecord.value != int(service.txt_record['ICD'])):
                    raise AttributeError(f'Not a known ICD type: {service.txt_record["ICD"]}')
```

Reply:
why are we not returning immediately as soon as we found something? Why keep looping to find other results?